cli/src/deploy/mod.rs

```diff
@@ -444,29 +455,29 @@ pub async fn add_deploy_config() {
     .interact_text()
     .unwrap();
   let mut cloud_configs = Vec::new();
-  let options = creds.keys().cloned().collect::<Vec<String>>();
-  if options.len() == 0 {
+  if creds.len() == 0 {
     prompt_new_cred(false).await
   }
+  let options = creds.keys().cloned().collect::<Vec<String>>();
   loop {
-    let selection = Select::new()
+    let selection = Select::with_theme(&ColorfulTheme::default())
```

Comment:
Oh, this isn't exactly what I meant. With this approach, you can't create new credentials during a deploy unless there are no credentials.

If you change it to be something like:

```rust
let mut options = creds.keys().cloned().collect::<Vec<String>>();
options.push("Create new credentials".to_string());
loop {
  let selection = Select::with_theme(&ColorfulTheme::default())...
  if selection == "Create new credentials" {
    prompt_new_cred(false).await;
    continue;
  }
  ...
```

Then if run with no credentials, the only option is "Create new credentials" where you can create the credentials and then it asks which credentials you want to use, or "Create new credentials" again.

That seems kinda silly, *but* if you're deploying another service after already having credentials for your first service, but then you remember that the new service needs different credentials, you can create them here without `Ctrl+C`-ing and creating them and then coming back here.

It's roughly the same LOC with a nicer experience, I think, and no less confusing.

Reply:
I see what you mean now